src/logic/function/basic.lean

```diff
@@ -540,14 +540,23 @@ along a function `f : α → β` to a function `β → γ`,
 by using the values of `g` on the range of `f`
 and the values of an auxiliary function `e' : β → γ` elsewhere.
 
-Mostly useful when `f` is injective. -/
+Mostly useful when `f` is injective,
+more generally when `f a  f b → g a = g b`. -/
 def extend (f : α → β) (g : α → γ) (e' : β → γ) : β → γ :=
 λ b, if h : ∃ a, f a = b then g (classical.some h) else e' b
 
 lemma extend_def (f : α → β) (g : α → γ) (e' : β → γ) (b : β) [decidable (∃ a, f a = b)] :
   extend f g e' b = if h : ∃ a, f a = b then g (classical.some h) else e' b :=
 by { unfold extend, congr }
 
+lemma extend_apply_of_unique (g : α → γ) (e' : β → γ)
+  (hf : ∀ (a b : α), f a = f b → g a = g b) (a : α) :
+  extend f g e' (f a) = g a :=
+begin
+  simp only [extend_def, dif_pos, exists_apply_eq_apply],
+  exact hf _ a (classical.some_spec (exists_apply_eq_apply f a)),
```

Comment:
Does
```lean
simpa only [extend_def, dif_pos, exists_apply_eq_apply] using hf _ a (classical.some_spec (exists_apply_eq_apply f a))
```
work?

Reply:
I don't understand, `simpa` does not work at all (“unknown identifier”).
Does it depend on some special file which should be opened?

Reply:
It's possible it's not imported here. I think it's better to not change the import of such a basic file, forget my comment.